handlers/home_assistant.py

```diff
@@ -0,0 +1,154 @@
+    
+    async def send_stock_alert(self, sku: str, price: str, url: str, in_stock: bool) -> None:
+        if not self.enabled or not self.connected:
+            return
+            
+        status = "IN STOCK" if in_stock else "OUT OF STOCK"
+        
+        notification_data = {
+            "title": "NVIDIA Stock Alert",
+            "message": f"{status}: {sku}\nPrice: {price}",
+            "target": self.notification_service,
+            "data": {
+                "url": url,
+                "tag": f"nvidia_stock_{sku}",
+                "color": "#00ff00" if in_stock else "#ff0000",
+                "priority": "high",
+                "sticky": True,
+                "actions": [
+                    {
+                        "action": "URI",
+                        "title": "View Product",
+                        "uri": url
+                    }
+                ]
+            }
+        }
+
+        await self._send_notification(notification_data)
+    
+    async def send_status_update(self, data: dict) -> None:
+        if not self.enabled or not self.connected:
+            return
+
+        status_text = "Successful" if data['last_check_success'] else "Failed"
+        
+        last_check_str = "No checks completed"
+        if data['last_check_time']:
+            last_check_str = data['last_check_time'].strftime("%H:%M:%S %d/%m/%Y")
+            minutes_since = data['time_since_check'].seconds // 60
+            last_check_str += f" ({minutes_since}m ago)"
+
+        message = (
+            f"Running for: {self.format_duration(data['runtime'])}\n"
+            f"Requests: {data['successful_requests']:,} successful, {data['failed_requests']:,} failed\n"
+            f"Last check: {last_check_str} ({status_text})\n"
+            f"Monitoring: {'None' if not data['monitored_cards'] else ', '.join(data['monitored_cards'])}"
+        )
+
+        notification_data = {
+            "message": message,
+            "title": "NVIDIA Stock Checker Status",
+            "data": {
+                "push": {
+                    "sound": "default",
+                    "priority": "normal"
+                },
+                "tag": "nvidia_stock_status",
+                "color": "#0099ff"
+            }
+        }
+
+        await self._send_notification(notification_data)
+    
+    async def send_startup_message(self, message: str) -> None:
+        if not self.enabled or not self.connected:
+            return
+            
+        notification_data = {
+            "message": message,
+            "title": "NVIDIA Stock Checker",
+            "data": {
+                "push": {
+                    "sound": "default",
+                    "priority": "normal"
+                },
+                "tag": "nvidia_stock_startup",
+                "color": "#0099ff"
+            }
+        }
+
+        await self._send_notification(notification_data)
+
+    def format_duration(self, duration):
```

Comment:
This could go into `notification.py` as I see it is reused across multiple files in `handlers`